src/CheckoutSdk/CheckoutConfiguration.cs

```diff
@@ -1,14 +1,39 @@
 namespace Checkout
 {
-    public class CheckoutConfiguration
+    public class CheckoutConfiguration : ICheckoutConfiguration
     {
-        public SdkCredentials SdkCredentials { get; }
+        public SdkCredentials SdkCredentials { get; private set; }
 
-        public Environment Environment { get; }
+        public Environment Environment { get; private set; }
 
-        public IHttpClientFactory HttpClientFactory { get; }
+        public IHttpClientFactory HttpClientFactory { get; private set; }
+
+        private CheckoutFilesConfiguration FilesApiConfiguration { get; set; }
+
+        public CheckoutConfiguration(
+            SdkCredentials sdkCredentials,
+            Environment environment,
+            IHttpClientFactory httpClientFactory)
+        {
+            Init(sdkCredentials, environment, httpClientFactory);
+            FilesApiConfiguration = CheckoutFilesConfiguration.Enabled(environment);
+        }
 
         public CheckoutConfiguration(SdkCredentials sdkCredentials, Environment environment,
+            IHttpClientFactory httpClientFactory, CheckoutFilesConfiguration filesApiConfiguration)
+        {
+            Init(sdkCredentials, environment, httpClientFactory);
+            FilesApiConfiguration = filesApiConfiguration;
+        }
+
+        public CheckoutFilesConfiguration GetFilesApiConfiguration()
+        {
+            return FilesApiConfiguration;
+        }
+
+        private void Init(
+            SdkCredentials sdkCredentials,
+            Environment environment,
             IHttpClientFactory httpClientFactory)
         {
```

Comment:
Method `Init` is not needed. Just assign the variables in the constructor and remove the redundant `private set;`s